From 43b4c23be4a2bcaa1ba06e179762d55c4cecfa96 Mon Sep 17 00:00:00 2001
From: Elliot Cameron <nixpkgs-commits@deshaw.com>
Date: Mon, 3 Mar 2025 16:47:59 -0500
Subject: [PATCH] qt5.qtwebengine: fix missing GN dependency

This missing dependency does not result in a build failure
in every case. Without the dependency, there is a race
condition in the build system that is only exposed when
the build can utilize more cores.

See https://bugs.gentoo.org/933368 for context.
---
 .../qtwebengine-fix_build_pdf_extension_util.patch    | 11 +++++++++++
 .../libraries/qt-5/modules/qtwebengine.nix            |  4 ++++
 2 files changed, 15 insertions(+)
 create mode 100644 pkgs/development/libraries/qt-5/modules/qtwebengine-fix_build_pdf_extension_util.patch

diff --git a/pkgs/development/libraries/qt-5/modules/qtwebengine-fix_build_pdf_extension_util.patch b/pkgs/development/libraries/qt-5/modules/qtwebengine-fix_build_pdf_extension_util.patch
new file mode 100644
index 00000000000000..f31c4a07092570
--- /dev/null
+++ b/pkgs/development/libraries/qt-5/modules/qtwebengine-fix_build_pdf_extension_util.patch
@@ -0,0 +1,11 @@
+# From https://933368.bugs.gentoo.org/attachment.cgi?id=896907
+--- ./src/3rdparty/chromium/qtwebengine/browser/pdf/BUILD.gn.orig	2024-07-03 13:49:13.812285886 +0100
++++ ./src/3rdparty/chromium/qtwebengine/browser/pdf/BUILD.gn	2024-07-03 13:51:45.789966941 +0100
+@@ -6,6 +6,7 @@ source_set("pdf") {
+   ]
+ 
+   deps = [
++    "//chrome/app:generated_resources",
+     "//content/public/browser",
+   ]
+ }
diff --git a/pkgs/development/libraries/qt-5/modules/qtwebengine.nix b/pkgs/development/libraries/qt-5/modules/qtwebengine.nix
index 48b343435d387b..40a7cace954a3c 100644
--- a/pkgs/development/libraries/qt-5/modules/qtwebengine.nix
+++ b/pkgs/development/libraries/qt-5/modules/qtwebengine.nix
@@ -157,6 +157,10 @@ qtModule (
         stripLen = 1;
         extraPrefix = "src/3rdparty/";
       })
+
+      # Fix race condition exposed by missing dependency
+      # https://bugs.gentoo.org/933368
+      ./qtwebengine-fix_build_pdf_extension_util.patch
     ];
 
     postPatch =
